tools/infrasys/src/s3.rs

```diff
@@ -0,0 +1,366 @@
+        formatted[..formatted.len() - 2].to_string()
+    } else {
+        formatted
+    }
+}
+
+/// Creates a *private* S3 Bucket using a CloudFormation template
+/// Input: The region in which the bucket will be created and the name of the bucket
+/// Output: The stack_arn of the stack w/ the S3 bucket, the CFN allocated bucket name,
+/// and the bucket url (for the url fields in Infra.lock)
+pub async fn create_s3_bucket(region: &str, stack_name: &str) -> Result<(String, String, String)> {
+    // TODO: Add support for accomodating pre-existing buckets (skip this creation process)
+    let cfn_client = CloudFormationClient::new(
+        Region::from_str(region).context(error::ParseRegion { what: region })?,
+    );
+    let cfn_filepath: PathBuf = format!(
+        "{}/infrasys/cloudformation-templates/s3_setup.yml",
+        shared::getenv("BUILDSYS_TOOLS_DIR")?
+    )
+    .into();
+    let cfn_template =
+        fs::read_to_string(&cfn_filepath).context(error::FileRead { path: cfn_filepath })?;
+    let stack_result = cfn_client
+        .create_stack(CreateStackInput {
+            stack_name: stack_name.to_string(),
+            template_body: Some(cfn_template.clone()),
+            ..Default::default()
+        })
+        .await
+        .context(error::CreateStack { stack_name, region })?;
+    // We don't have to wait for successful stack creation to grab the stack ARN
+    let stack_arn = stack_result
+        .clone()
+        .stack_id
+        .context(error::ParseResponse {
+            what: "stack_id",
+            resource_name: stack_name,
+        })?;
+
+    // Grab the StackOutputs to get the Bucketname and BucketURL
+    let output_array = shared::get_stack_outputs(&cfn_client, &stack_name, region).await?;
+    let bucket_name = output_array[0]
+        .output_value
+        .as_ref()
+        .context(error::ParseResponse {
+            what: "outputs[0].output_value (bucket name)",
+            resource_name: stack_name,
+        })?
+        .to_string();
+    let bucket_rdn = output_array[1]
+        .output_value
+        .as_ref()
+        .context(error::ParseResponse {
+            what: "outputs[1].output_value (bucket url)",
+            resource_name: stack_name,
+        })?
+        .to_string();
+
+    Ok((stack_arn, bucket_name, bucket_rdn))
+}
+
+/// Adds a BucketPolicy allowing GetObject access to a specified VPC
+/// Input: Region, Name of bucket, which prefix root.json should be put under, and vpcid
+/// Note that the prefix parameter must have the format "/<folder>/*" and the bucket name "<name>"
+/// Output: Doesn't need to save any metadata from this action  
+pub async fn add_bucket_policy(
+    region: &str,
+    bucket_name: &str,
+    prefix: &str,
+    vpcid: &str,
+) -> Result<()> {
+    // Get old policy
+    let s3_client =
+        S3Client::new(Region::from_str(region).context(error::ParseRegion { what: region })?);
+    let mut current_bp: serde_json::Value = match s3_client
```

Comment:
Nit: once you amend it, calling it 'current` doesn't quite work. And abbreviating with `bp` obscures what it represents. (I think the same change would be nice in the tests.)

```suggestion
    let mut policy: serde_json::Value = match s3_client
```